network/p2p/peerManager_test.go

```diff
@@ -218,9 +240,32 @@ func (suite *PeerManagerTestSuite) TestConcurrentOnDemandPeerUpdate() {
 	}, 10*time.Second, 100*time.Millisecond)
 }
 
-// assertListsEqual asserts that two identity list are equal ignoring the order
-func assertListsEqual(t *testing.T, list1, list2 flow.IdentityList) {
-	list1 = list1.Sort(order.ByNodeIDAsc)
-	list2 = list2.Sort(order.ByNodeIDAsc)
-	assert.Equal(t, list1, list2)
+// assertListsEqual asserts that two peer ID slices are equal ignoring the order
+func assertListsEqual(t *testing.T, list1, list2 peer.IDSlice) {
+	listsEqual := func() bool {
+		if len(list1) != len(list2) {
+			return false
+		}
+
+		map1 := make(map[peer.ID]int)
+		map2 := make(map[peer.ID]int)
+
+		for _, e1 := range list1 {
+			map1[e1]++
+		}
+		for _, e2 := range list2 {
+			map2[e2]++
+		}
+
+		for key, val := range map1 {
+			if map2[key] != val {
+				return false
+			}
+		}
+		return true
+	}
+
+	if !listsEqual() {
+		assert.Failf(t, "peer ID slices not equal", "list1: %v, list2: %v", list1, list2)
+	}
```

Comment:
I've recently discovered [`assert.Subset`](https://pkg.go.dev/github.com/stretchr/testify/assert#Assertions.Subset)

Reply:
How did I not find this haha, I even specifically googled for it cause i was sure something like this must exist.